src/data_factory_testing_framework/functions/evaluator/rules/logical_bool_expression_rule.py

```diff
@@ -0,0 +1,67 @@
+from typing import Union
+
+from lark import Tree
+
+from data_factory_testing_framework.exceptions.expression_evaluation_error import ExpressionEvaluationError
+from data_factory_testing_framework.functions.evaluator.exceptions import (
+    ExpressionEvaluationInvalidChildTypeError,
+    ExpressionEvaluationInvalidNumberOfChildrenError,
+)
+from data_factory_testing_framework.functions.evaluator.rules.expression_rule import EvaluatedExpression
+
+from .expression_rule import ExpressionRuleEvaluator
+
+
+class LogicalBoolExpressionEvaluator(ExpressionRuleEvaluator):
+    OR = "or"
+    AND = "and"
+
+    def __init__(self, tree: Tree) -> None:
+        """Initializes the expression rule evaluator."""
+        super().__init__(tree)
+
+        if len(self.children) != 3:
+            raise ExpressionEvaluationInvalidNumberOfChildrenError(required=3, actual=len(self.children))
+
+        for i, child in enumerate(self.children):
+            self._check_child_type(child, i)
+
+        if self.children[0].value not in (self.OR, self.AND):
+            self._raise_invalid_operator(self.children[0].value)
+
+    def evaluate(self) -> EvaluatedExpression:
+        logical_operator: EvaluatedExpression = self.children[0]
+        left_expression = self.children[1]
+        right_expression = self.children[2]
+
+        if logical_operator.value == self.OR:
+            result = self._evaluate_expression(left_expression) or self._evaluate_expression(right_expression)
+        elif logical_operator.value == self.AND:
+            result = self._evaluate_expression(left_expression) and self._evaluate_expression(right_expression)
+        else:
+            self._raise_invalid_operator(logical_operator.value)
+        return EvaluatedExpression(value=result)
+
+    def _raise_invalid_operator(self, logical_operator: str) -> None:
+        raise ExpressionEvaluationError(f"Invalid logical operator: {logical_operator}")
+
+    def _check_child_type(self, child: Union[EvaluatedExpression, ExpressionRuleEvaluator], child_index: int) -> None:
+        if not isinstance(child, (ExpressionRuleEvaluator, EvaluatedExpression)):
+            raise ExpressionEvaluationInvalidChildTypeError(
+                child_index=child_index,
+                expected_types=(ExpressionRuleEvaluator, EvaluatedExpression),
+                actual_type=type(child),
+            )
+
+    def _evaluate_expression(self, expression: Union[ExpressionRuleEvaluator, EvaluatedExpression]) -> bool:
+        if isinstance(expression, ExpressionRuleEvaluator):
```

Comment:
I noticed that the pattern of instance-checking RuleEvaluator and EvaluatedExpression and then either calling evaluate() or .value respectively is happening a lot. Perhaps we can abstract this away in a single place, hopefully in the class itself?

Reply:
We are getting there :100: ! Named it for now `ensure_evaluated_expression` and let's find a better name once we also adjust `EvaluatedExpression`.